src/core/pointcloud/qgspointclouddataprovider.h

```diff
@@ -189,6 +200,23 @@ class CORE_EXPORT QgsPointCloudDataProvider: public QgsDataProvider
      */
     virtual QgsGeometry polygonBounds() const;
 
+    /**
+     * Triggers generation of the point cloud index
+     *
+     * emits statisticsGenerationStateChanged()
+     *
+     * \sa statsCalculator()
+     */
+    void generateStatistics();
```

Comment:
It would be good to define semantics of this function when:
- stats have been already calculated
- stats are being calculated

I guess it also deserves a bit more explanation about how it works - is this starting the generation in a background thread? when does it emit the signal - immediately, or later?